File: 1]. DSA/1]. Data Structures/01]. Array/JavaScript/_018)_Leaders_in_an_Array.js
//if current element/person value is greater than front element/person then current element is considered as a leader
function leaders(arr) {
    let n = arr.length;
    let cur_leader = arr[n - 1];
    let leaders = [arr[n - 1]];
    for (let i = n - 2; i >= 0; i--) {
        if (cur_leader < arr[i]) {
            cur_leader = arr[i];
            leaders.push(cur_leader);
        }
    }
    return leaders;
}

let arr = [7, 10, 4, 3, 6, 5, 2];
console.log(`Leaders = ${leaders(arr)}`);

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_018)_Leaders_in_an_Array.dart
/*
If current element/person value is greater than front element/person 
then current element is considered as a leader
*/

List findLeaders(arr) {
  var n = arr.length;
  var currLeader = arr[n - 1];
  var leaders = [arr[n - 1]];

  for (int i = n - 2; i >= 0; i--) {
    if (arr[i] > currLeader) {
      currLeader = arr[i];
      leaders.add(currLeader);
    }
  }
  return leaders;
}

void main() {
  var arr = [7, 10, 4, 3, 6, 5, 2];
  var leaders = findLeaders(arr);
  print('Leaders in the array are: $leaders');
}
